Feature/cldsrv 223 implement putbuckettagging

williamlardier opened the pull request:
Cherry-picked the first two commits from https://github.com/scality/cloudserver/pull/4451.

Added some new tests.

**Monitoring for the API is added in https://github.com/scality/cloudserver/pull/4594**

Commit:
CLDSRV-167: implement put bucket tagging

bert-e commented:
# Hello williamlardier,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/cloudserver/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-223 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.5.3`


Please check the `Fix Version/s` of CLDSRV-223, or the target
branch of this pull request.

Commit:
CLDSRV-223: Implement bucket owner check and change check order
Checking order has been changed to fit with AWS one, meaning that we
raise the first error we face

- Checking every tag one by one
- Checking that the tag key is unique, the error is raised only if we
  already seen a tag with the same key before
- Checking that the key length is less than 128
- Checking that the tag value length is less than 256
- After seeing more than 50 tags, if there is one more, raise the too
  many tags error

Seragonia committed:
CLDSRV-223: Fix comparison and add tests

Seragonia committed:
CLDSRV-223: Bump Arsenal

Commit:
CLDSRV-167: use arsenal parse tag function

XinLi0207 commented on eve/main.yml:
```diff
@@ -92,7 +92,7 @@ models:
   - ShellCommand: &yarn-install
       name: install modules
-      command: yarn install --ignore-engines --frozen-lockfile --network-concurrency 1
+      command: yarn install --ignore-engines --frozen-lockfile --network-concurrency=1
```
here should we use `=` or space?

williamlardier replied:
I am guessing both are valid, but as this was changed ion 7.10, to stay consistent, I wanted to keep this change while cherry-picking

XinLi0207 reviewed (approved):
LGTM

ghost commented:
/approve

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-223 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.5.4`


Please check the `Fix Version/s` of CLDSRV-223, or the target
branch of this pull request.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.5`



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.10`
* `development/7.4`
* `development/8.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.5`



The following branches have **NOT** changed:

* `development/6.4`
* `development/7.10`
* `development/7.4`
* `development/8.4`



Please check the status of the associated issue CLDSRV-223.

Goodbye williamlardier.
